src/dataflowd/src/lib.rs

```diff
@@ -145,27 +153,51 @@ impl<S: Client> Client for SplitClient<S> {
 
         Ok(())
     }
-    async fn recv(&mut self) -> Option<Response> {
-        // TODO: We currently don't have a good way to receive from many clients
-        let mut futures = FuturesUnordered::new();
-        for client in self.compute_clients.values_mut() {
-            if let ComputeClientFlavor::Remote(client) = client {
-                futures.push(client.recv());
+}
+
+impl<C: Client> Stream for SplitClient<C> {
+    type Item = Response;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = &mut *self;
+
+        // Poll {each of the compute clients, the storage client} in turn.
+        for _ in 0..=this.compute_clients.len() {
+            match this.compute_clients.get_mut(&this.recv_cursor) {
+                None => {
```

Comment:
These aren't likely to be contiguous, so I think this check is not right. Not trivial to fix afaict. Maybe a `BTreeMap` and picking an offset in the enumeration?

Reply:
Yeah, `BTreeMap` seems like the way, though oy the implementation is painful. Seems correct, though. Thanks very much for the spot!